src/OrchardCore.Modules/OrchardCore.AdminMenu/AdminNodes/PlaceholderAdminNodeDriver.cs

```diff
@@ -18,44 +17,49 @@ public PlaceholderAdminNodeDriver(IAdminMenuPermissionService adminMenuPermissio
             _adminMenuPermissionService = adminMenuPermissionService;
         }
 
-        public override IDisplayResult Display(PlaceholderAdminNode treeNode)
+        public override Task<IDisplayResult> DisplayAsync(PlaceholderAdminNode treeNode, BuildDisplayContext context)
```

Comment:
@Piedone we had the same argument during the last meeting. @sebastienros argument was following the Microsoft standard. Microsoft never implement sync and async version of the same method. Which kinda makes sense. As you see I exposed CombineAsync method to reduce the code written to leave only one method which will be the best was for teaching someone and also for performance.

Reply:
Oh, and if we want to do that, we'll have to introduce a breaking change to add the sync versions that accept the context.

Reply:
It's frequent in .NET to have both though, like all the file I/O (including methods introduced in later .NET versions) or `IDistributedCache`. But that aside, my point is that if the consumers of our API will frequently write only sync code, which is apparently the case, then we should ideally make that easier (and/or even consider `ValueTask`, but I don't think we can have enough data to rule for or against that). `CombinedAsync` is indeed good, but there's still a lot of awkward `Task.FromResult` still.

But I see the issue with the call chain. There must be a way out of this somehow, this looks simple... The only one I can think of though is adding a condition to each override like this:

```c#
public virtual async Task<IDisplayResult> DisplayAsync(TModel model, TDisplayContext context)
{
    return Display(model, context) ?? await DisplayAsync(model, context.Updater);
}

public virtual IDisplayResult Display(TModel model, TDisplayContext context)
{
    return Display(model, context.Updater);
}

...
```

But this is ugly and adds branching to something called very frequently.

Or, we accept that this won't be great in one go, merge this PR, and introduce an `IDisplayResult Display(TModel model, TDisplayContext context)` only in 3.0.

Did you figure out something better today? Hence https://github.com/OrchardCMS/OrchardCore/pull/16508#issuecomment-2263655591 @sebastienros?

@sarahelsaig as Lombiq's Chief Orchard Core Upgrade Officer, could you please take a look and provide feedback?

Reply:
I approve of marking all the sync versions obsolete. It keeps the API simpler and I still remember that having both overridables in the base class was confusing back when I started working with OC.

I don't think returning `Task.FromResult()` is not a huge ask. Could we add an `abstract class SyncDisplayDriver<TModel, TDisplayContext, TEditorContext, TUpdateContext> : DisplayDriver<TModel, TDisplayContext, TEditorContext, TUpdateContext>` type to OrchardCore.DisplayManagement? That would be a nice way to bridge the gap: when you are sure you are only doing sync stuff in a driver you can inherit from it instead of the base `DisplayDriver`. If you need both sync and async methods in the same driver, then you can put up with the extra `Task.FromResult()`.

Reply:
Hmm, having a fully sync base driver is an interesting idea! Though note that `DisplayDriver<TModel, TDisplayContext, TEditorContext, TUpdateContext>` is only one of the base classes, and while lower-level drivers directly inherit from it, we also have `ContentPartDisplayDriver<TPart>` and `ContentFieldDisplayDriver<TField>`. Those have similar issues but seem to be more consistent with offering sync overloads.

Reply:
Yeah, in the linked PR I've also included sync versions is both of the classes you mentioned, and altered a few classes to use them.

Reply:
Didn't you forget to push? Because I don't see part or field drivers there.

Reply:
@Piedone @sebastienros @sarahelsaig please check out the updated. Now, every important async method has a corresponding sync method. When possible, I marked old methods as obsolete. However, it was necessary to remove some methods to accomplish this.

I did the same change to the `SectionDisplayDriver` class. Also, simplified the site settings by introducing a new base class `SiteDisplayDriver<SomeSettings>`. 

Please review the changes and provide feedback.

Reply:
@Piedone  I think @sarahelsaig was talking about another draft PR https://github.com/OrchardCMS/OrchardCore/pull/16514.

Personally, I like to use a single driver for this so we don't add any more confusion. I think having one sync and one async version is good approach for this.

Reply:
Yep, I've check that PR. Anyway, yeah, all in all single base drivers would be better.

Reply:
Good I think we are all on the same page. Can you please give this PR a review hoping to merge it soon.

Reply:
Yeah, on it, just it's quite a lot.

Reply:
Yes it is alot. Obviously everything is important, but the key changes to review is `SectionDisplayDriver` and `DisplayDriver`

Reply:
I am resolving this since we all agree on the approach.